src/java.base/share/classes/java/lang/Process.java

```diff
@@ -149,12 +169,93 @@ public abstract class Process {
     private Charset inputCharset;
     private BufferedReader errorReader;
     private Charset errorCharset;
+    private boolean closed;     // true if close() has been called
 
     /**
      * Default constructor for Process.
      */
     public Process() {}
 
+    /**
+     * Close all writer and reader streams and terminate the process.
+     * The streams are closed immediately and the process is terminated without waiting.
+     * This method is idempotent, if the process has already been closed
+     * invoking this method has no effect.
+     * <p>
+     * If the data from the process streams is needed, it must be read before calling this method.
+     * The contents of streams that have not been read fully are lost, they are discarded or ignored.
+     * <p>
+     * If the process should be allowed to run to completion or the exit value of the process
+     * is of interest, then the caller must {@linkplain #waitFor() wait} for the process
+     * to complete before calling this method.
+     * Calling {@link #waitFor() waitFor} before calling {@code close} or exiting
+     * the try-with-resources block allows the process time to clean up and exit normally.
+     * <p>
+     * Streams should be closed when no longer needed.
+     * Closing an already closed stream usually has no effect but is specific to the stream.
+     * If an {@code IOException} occurs when closing a stream it is
+     * re-thrown after the process is terminated. Additional {@code IOExceptions}
+     * thrown by closing the remaining streams, if any, are added to the first
+     * {@code IOException} as {@linkplain IOException#addSuppressed suppressed exceptions}.
+     * <p>
+     * The process may already have terminated or be in the process of terminating;
+     * if it is {@linkplain #isAlive() alive}, it is {@linkplain #destroy destroyed}.
```

Comment:
The compound sentence is hard to read. It might be clear to say that the process is destroyed. This is no-op if the process has already terminated.